contracts/exchangeIssuance/FlashMintLeveragedExtended.sol

```diff
@@ -0,0 +1,590 @@
+        DEXAdapter.SwapData memory _swapDataDebtForCollateral,
+        DEXAdapter.SwapData memory _swapDataInputToken
+    )
+        external
+        override
+        payable
+        nonReentrant
+    {
+        uint256 inputTokenBalanceBefore = IERC20(addresses.weth).balanceOf(address(this));
+        IWETH(addresses.weth).deposit{value: msg.value}();
+        _initiateIssuance(
+            _setToken,
+            _setAmount,
+            DEXAdapter.ETH_ADDRESS,
+            msg.value,
+            _swapDataDebtForCollateral,
+            _swapDataInputToken
+        );
+        uint256 amountToReturn = IERC20(addresses.weth).balanceOf(address(this)).sub(inputTokenBalanceBefore);
+        IWETH(addresses.weth).withdraw(amountToReturn);
+        msg.sender.transfer(amountToReturn);
+    }
+
+    function issueSetFromExactERC20(
+        ISetToken _setToken,
+        uint256 _minSetAmount,
+        address _inputToken,
+        uint256 _inputTokenAmount,
+        DEXAdapter.SwapData memory _swapDataDebtForCollateral,
+        DEXAdapter.SwapData memory _swapDataInputToken,
+        uint256 _priceEstimateInflator,
+        uint256 _maxDust
+    )
+        external
+        nonReentrant
+        returns(uint256)
+    {
+        IERC20(_inputToken).transferFrom(msg.sender, address(this), _inputTokenAmount);
+
+        return _issueSetFromExactInput(
+            _setToken,
+            _minSetAmount,
+            _inputToken,
+            _inputTokenAmount,
+            _swapDataDebtForCollateral,
+            _swapDataInputToken,
+            _priceEstimateInflator,
+            _maxDust
+        );
+    }
+
+    function issueSetFromExactETH(
+        ISetToken _setToken,
+        uint256 _minSetAmount,
+        DEXAdapter.SwapData memory _swapDataDebtForCollateral,
+        DEXAdapter.SwapData memory _swapDataInputToken,
+        uint256 _priceEstimateInflator,
+        uint256 _maxDust
+    )
+        external
+        payable
+        nonReentrant
+        returns(uint256)
+    {
+        IWETH(addresses.weth).deposit{value: msg.value}();
+        return _issueSetFromExactInput(
+            _setToken,
+            _minSetAmount,
+            DEXAdapter.ETH_ADDRESS,
+            msg.value,
+            _swapDataDebtForCollateral,
+            _swapDataInputToken,
+            _priceEstimateInflator,
+            _maxDust
+        );
+    }
+
+    function _issueSetFromExactInput(
+        ISetToken _setToken,
+        uint256 _minSetAmount,
+        address _inputToken,
+        uint256 _inputTokenAmount,
+        DEXAdapter.SwapData memory _swapDataDebtForCollateral,
+        DEXAdapter.SwapData memory _swapDataInputToken,
+        uint256 _priceEstimateInflator,
+        uint256 _maxDust
+    )
+        internal
+        returns(uint256)
+    {
+        require(_inputTokenAmount > _maxDust, "FlashMintLeveragedExtended: _inputToken must be more than _maxDust");
+
+        while (_inputTokenAmount > _maxDust) {
```

Comment:
can we replace this while loop with a contract level parameter `MAX_ITERATIONS` check?

either way if the params of the call are proper then this value shouldn't be hit. lets be sure to add natspec clarifying